code/upaya/config/initializers/saml_idp.rb

```diff
@@ -0,0 +1,60 @@
+require 'service_provider'
+
+SamlIdp.configure do |config|
+  protocol = Rails.env.development? ? 'http://' : 'https://'
+  api_base = protocol + Figaro.env.domain_name + '/api'
+
+  config.x509_certificate = Rails.application.secrets.saml_cert
+  config.secret_key = OpenSSL::PKey::RSA.new(
+    File.read(Rails.root + 'config/saml.key.enc'),
+    Figaro.env.saml_passphrase).to_pem
+
+  # config.signature_alg = 'rsa-sha256'
+  # config.digest_alg = 'sha256'
+
+  # Disabled because authnrequest signing doesn't prevent any
+  # attack that can be thought of.
+  # config.verify_authnrequest_sig = true
```

Comment:
Totally, it's a really good phishing vector since the victim clicks a link to a trusted source (the IdP) and scanning engines are most likely domain based and not concerned about the parameters of the request (which could contain a malicious redirect, for example, mirror the IdP's pw reset form). 

🐟